src/palladio/SOPAssign.cpp

```diff
@@ -554,8 +566,16 @@ void SOPAssign::buildUI(GU_Detail* detail, ShapeData& shapeData, const ShapeConv
 		const std::wstring attrName = ra.niceName;
 		const std::wstring attrId = toUTF16FromOSNarrow(NameConversion::toPrimAttr(ra.fqName).toStdString());
 
-		const auto annotationInfo = AnnotationParsing::getAttributeAnnotationInfo(ra.fqName, ruleFileInfo);
-		const std::wstring description = annotationInfo.mDescription;
+		const auto& annotationsIt = attributeAnnotations.find(ra.fqName);
+		if (annotationsIt == attributeAnnotations.end())
+			continue;
+		auto& traitAnnotationMap = annotationsIt->second;
+
+		const auto& enumIt = traitAnnotationMap.find(AnnotationParsing::AttributeTrait::ENUM);
+		const auto& descriptionIt = traitAnnotationMap.find(AnnotationParsing::AttributeTrait::DESCRIPTION);
```

Comment:
too many `auto`s for my taste, maybe introducing type aliases for the `attributeAnnotations` map type?

Reply:
These are all iterators returned by "map.find()"
I replaced the map type in the header and cpp of AnnotationParsing aswell as the one time it's used in SOPAssign.